storage/s3_test.go

```diff
@@ -128,6 +128,97 @@ func TestNewSessionWithNoSignRequest(t *testing.T) {
 	}
 }
 
+func TestNewSessionWithProfileFromFile(t *testing.T) {
+	// create a temporary credentials file
+	file, err := os.CreateTemp("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(file.Name())
+
+	profiles := `[default]
+aws_access_key_id = default_profile_key_id
+aws_secret_access_key = default_profile_access_key
+
+[p1]
+aws_access_key_id = p1_profile_key_id
+aws_secret_access_key = p1_profile_access_key
+
+[p2]
+aws_access_key_id = p2_profile_key_id
+aws_secret_access_key = p2_profile_access_key
+
+[p3]
+aws_access_key_id = p3_profile_key_id
+aws_secret_access_key = p3_profile_access_key`
+
+	_, err = file.Write([]byte(profiles))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testcases := []struct {
+		fileName           string
+		profileName        string
+		expAccessKeyId     string
+		expSecretAccessKey string
+	}{
+		{
+			fileName:           file.Name(),
+			profileName:        "",
+			expAccessKeyId:     "default_profile_key_id",
+			expSecretAccessKey: "default_profile_access_key",
+		},
+		{
+			fileName:           file.Name(),
+			profileName:        "p1",
+			expAccessKeyId:     "p1_profile_key_id",
+			expSecretAccessKey: "p1_profile_access_key",
+		},
+		{
+			fileName:           file.Name(),
+			profileName:        "non-existent-profile",
+			expAccessKeyId:     "",
+			expSecretAccessKey: "",
+		},
+		{
+			fileName:           file.Name(),
+			profileName:        "p2",
+			expAccessKeyId:     "p2_profile_key_id",
+			expSecretAccessKey: "p2_profile_access_key",
+		},
+		{
+			fileName:           file.Name(),
+			profileName:        "p3",
+			expAccessKeyId:     "p3_profile_key_id",
+			expSecretAccessKey: "p3_profile_access_key",
+		},
+	}
+	for _, tc := range testcases {
```

Comment:
Could you please add a descriptive name for all the test cases and run each test with `t.Run` as in https://github.com/peak/s5cmd/blob/master/storage/s3_test.go#L78 ?